cli/src/check.rs

```diff
@@ -0,0 +1,195 @@
+use anyhow::Result;
+use k8s_openapi::{
+    api::{apps::v1::Deployment, core::v1::Namespace},
+    apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition,
+};
+use kube::{api::ListParams, Api, Client, ResourceExt};
+use serde::Serialize;
+
+const CHECK: &str = "√";
+const EX: &str = "×";
+
+#[derive(Serialize)]
+struct CheckOutput {
+    success: bool,
+    categories: Vec<Category>,
+}
+
+#[derive(Serialize)]
+struct Category {
+    category_name: &'static str,
+    checks: Vec<CheckResult>,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "lowercase")]
+enum CheckStatus {
+    Success,
+    Error,
+}
+
+#[derive(Serialize, Default)]
+pub struct CheckResult {
+    description: &'static str,
+    result: CheckStatus,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    hint: Option<&'static str>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    error: Option<String>,
+}
+
+impl Default for CheckStatus {
+    fn default() -> CheckStatus {
+        CheckStatus::Success
+    }
+}
+
+impl CheckResult {
+    fn new(description: &'static str) -> CheckResult {
+        CheckResult {
+            description,
+            ..Default::default()
+        }
+    }
+}
+
+pub async fn check(client: Client, pre: bool) -> Vec<CheckResult> {
```

Comment:
I'd be inclined to split this function into smaller functions, each of which return a `CheckResult`